Fix aircraft idle speed and turn speed inconsistencies

reaperrr opened the pull request:
Fixes #19278.

reaperrr committed:
Fix IdleTurnSpeed ignoring trait pause/disable

reaperrr committed:
Minor FlyIdle perf optimization

None of these prerequisites change on the fly,
so cache the result in the activity ctor.

reaperrr committed:
Introduce IdleMovementSpeed

That actually factors in speed modifiers and trait pause/disable.

reaperrr committed:
Improve Aircraft TurnSpeed getters readability

reaperrr committed:
Make aircraft turn speed scale with speed modifiers

pchote commented on OpenRA.Mods.Common/Traits/Air/Aircraft.cs:
```diff
@@ -268,6 +268,19 @@ public WAngle Roll
+
+			var turnSpeed = isIdleTurn ? IdleTurnSpeed ?? TurnSpeed : TurnSpeed;
+
+			turnSpeed = new WAngle(Util.ApplyPercentageModifiers(turnSpeed.Angle, speedModifiers).Clamp(1, 1024));
```
`return new WAngle(...)`.

pchote reviewed (commented):
Fixes confirmed, and LGTM overall.

Just two small nits:
* The compiler is smart enough that the "Aircraft turn speed getters perf optimization" commit shouldn't change anything in practice

abcdefg30 commented:
> The compiler is smart enough that the "Aircraft turn speed getters perf optimization" commit shouldn't change anything in practice

Fwiw I find it easier to read the new order (||).

reaperrr commented:
Updated.

> Fwiw I find it easier to read the new order (||).

Same here, that was part of the reason I made this change. Updated commit title/description to reflect this.

pchote commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/e30e8ac389c0f637828690802c800715f679cb72
